src/ostree/ot-admin-builtin-pin.c

```diff
@@ -54,32 +119,23 @@ ot_admin_builtin_pin (int argc, char **argv, OstreeCommandInvocation *invocation
       char *endptr = NULL;
 
       errno = 0;
-      const guint64 deploy_index = g_ascii_strtoull (deploy_index_str, &endptr, 10);
+      gint64 deploy_index = g_ascii_strtoull (deploy_index_str, &endptr, 10);
       if (*endptr != '\0')
-        return glnx_throw (error, "Invalid index: %s", deploy_index_str);
-      if (errno == ERANGE)
+        {
+          if (!g_strcmp0 (deploy_index_str, "booted"))
+            deploy_index = get_deployment_index_for_type (sysroot, OSTREE_DEPLOYMENT_BOOTED);
+          else if (!g_strcmp0 (deploy_index_str, "pending"))
+            deploy_index = get_deployment_index_for_type (sysroot, OSTREE_DEPLOYMENT_PENDING);
+          else if (!g_strcmp0 (deploy_index_str, "rollback"))
+            deploy_index = get_deployment_index_for_type (sysroot, OSTREE_DEPLOYMENT_ROLLBACK);
+
```

Comment:
Feels like we should do something to explicitly handle the notfound aka `-1` case here?  Like:

```
if (deploy_index < 0)
  return glnx_throw (error, "Deployment type not found: %s", deploy_index_str);
```

Reply:
It's already handled inside ot_admin_get_indexed_deployment function, so I didn't worry about handling it again, but we can handle it here also if needs be.

Reply:
That will tell the user `error: Invalid index -1` which seems much less friendly than `error: Deployment type not found: rollback` right?